### PR TITLE
New package: opencl-amd-18.50.708488_1

### PR DESCRIPTION
**Pull request by buech**
Requested in #791 and based on the Arch Linux [PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=opencl-amd).

I am unsure about whether this is allowed in the repos though, because the [EULA](https://www.amd.com/en/support/eula) says:

> 3. RESTRICTIONS
Except for the limited license expressly granted in Section 2 herein, You have
no other rights in the Software, whether express, implied, arising by estoppel
or otherwise. Further restrictions regarding Your use of the Software are set
forth below. You may not:
> - modify or create derivative works of the Software;
> - distribute, assign or otherwise transfer the Software;
> - decompile, reverse engineer, disassemble or otherwise reduce the Software to
a human-perceivable form (except as allowed by applicable law);
> - alter or remove any copyright, trademark or patent notice(s) in the Software;
or
> - use the Software to: (i) develop inventions directly derived from
confidential information to seek patent protection; (ii) assist in the
analysis of Your patents and patent applications; or (iii) modify existing
patents.;
> - use, modify and/or distribute any of the Software so that any part becomes
subject to a Free Software License.

### CODE REVIEW

**Comment by Hoshpak** [comment]
This is clearly non-distributable. It could only be accepted if set to restricted.

File: srcpkgs/opencl-amd/template [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+short_desc="OpenCL userspace driver as provided by the amdgpu-pro driver stack"
+maintainer="Adam Buechner <adi51@gmx.de>"
+license="custom"
+homepage="http://www.amd.com"
+archs="x86_64"
+repository=nonfree
+
+_prefix="amdgpu-pro-"
+_postfix="-ubuntu-18.04"
+_major="18.50"
```
Comment (Chocimier) [inline srcpkgs/opencl-amd/template]: Please do it like here: https://github.com/void-linux/void-packages/blob/master/srcpkgs/ebtables/template#L5 to keep version defined once.

File: srcpkgs/opencl-amd/template [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+_minor="708488"
+
+distfiles="https://drivers.amd.com/drivers/linux/${_prefix}${_major}-${_minor}${_postfix}.tar.xz"
+checksum="b7b7bc9796e524e35f1f3ccbd99ba89cea383a0ed5a5357084522da0bdc09d59"
+
+XBPS_FETCH_CMD="wget --referer https://support.amd.com/en-us/kb-articles/Pages/AMDGPU-PRO-Driver-for-Linux-Release-Notes.aspx -N"
```
Comment (Chocimier) [inline srcpkgs/opencl-amd/template]: Try to export [HTTP_REFERER](https://github.com/void-linux/void-packages/pull/7514#issuecomment-456050166) instead.

File: srcpkgs/opencl-amd/template [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	popd
+	popd
+
+	mkdir libdrm
+	pushd libdrm
+	ar x "${wrksrc}/${_prefix}${_major}-${_minor}${_postfix}/libdrm-amdgpu-amdgpu1_2.4.95-${_minor}_amd64.deb"
+	tar xJf data.tar.xz
+	pushd opt/amdgpu/lib/x86_64-linux-gnu
+	rm libdrm_amdgpu.so.1
+	mv libdrm_amdgpu.so.1.0.0 libdrm_amdgpo.so.1.0.0
```
Comment (Chocimier) [inline srcpkgs/opencl-amd/template]: Why is rename needed?

Comment (buech) [reply srcpkgs/opencl-amd/template]: It seems libdrm_amdgpu.so is already provided by another package but the library from the amdgpu-pro package is still needed for OpenCL.

File: srcpkgs/opencl-amd/template [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+pkgname=opencl-amd
+version=18.50.708488
+revision=1
+create_wrksrc=yes
+depends="ocl-icd"
+short_desc="OpenCL userspace driver as provided by the amdgpu-pro driver stack"
+maintainer="Adam Buechner <adi51@gmx.de>"
+license="custom"
+homepage="http://www.amd.com"
+archs="x86_64"
```
Comment (maxice8) [inline srcpkgs/opencl-amd/template]: archs should go directly above create_wrksrc=

**Comment by Cymity** [comment]
Would it be possible to update this to 19.10?

**Comment by buech** [comment]
I don't have access to my workstation at the moment so I can't test
that. I would appreciate if anyone else could try it.

**Comment by ericonr** [comment]
Ping here. Do we still want this package?

**Comment by ericonr** [comment]
Closing due to inactivity and general "unpackageability".

Feel free to reopen if you're still interested in maintaining this.

**Pull request closed without merging**